src/shared/assertUnreachable.ts

```diff
@@ -0,0 +1,5 @@
+// @see https://www.typescriptlang.org/docs/handbook/advanced-types.html#exhaustiveness-checking
+/* istanbul ignore next */
+export function assertUnreachable(value: never): never {
+  throw new TypeError(`Unreachable value reached ${value}`);
+}
```

Comment:
I'm glad you separated this out. I was going to ask you to add this to the excluded files in rollup so we don't export an `assertUnreachable` file, but our config already handles this 😃

https://github.com/apollographql/space-kit/blob/b3fa01c71208ae0e1f0e15b32e85cf846b80a142/rollup.config.js#L32-L46